src/field_theory/krull_topology.lean

```diff
@@ -283,3 +285,219 @@ begin
+  end,
+  map_add' := λ x y,
+  begin
+    let E := intermediate_field.adjoin K ({x, y} : set L),
+    have h_sub := intermediate_field.gc.le_u_l {x, y},
+    have hx := h_sub (mem_insert x _),
+    have hy := h_sub (mem_insert_of_mem _ $ mem_singleton _),
+    have hE : finite_dimensional K E := intermediate_field.adjoin.finite_dimensional_of_finite_set
+    ((finite_singleton _).insert _) (λ a _, h_int a),
+    rw [f.glued_generators_of_pushforwards_function_spec h_int hE hx,
+      f.glued_generators_of_pushforwards_function_spec h_int hE hy,
+      f.glued_generators_of_pushforwards_function_spec h_int hE (E.add_mem hx hy)],
+    exact map_add (f.generator_of_pushforward hE) ⟨x, hx⟩ ⟨y, hy⟩,
+  end,
+  commutes' := λ r,
+  begin
+    rw [f.glued_generators_of_pushforwards_function_spec h_int
+      (intermediate_field.finite_dimensional_bot K L)
+      (intermediate_field.algebra_map_mem _ r)],
+    exact (f.generator_of_pushforward _).commutes r,
+  end }
+
+lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_injective
+  (h_int : algebra.is_integral K L)
+  (f : ultrafilter (L →ₐ[K] L)) : function.injective
+  (ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int) :=
+(ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int).to_ring_hom.injective
+
+lemma eq_of_map_le {f : L →ₐ[K] L} (h_findim : finite_dimensional K E) (h_map_le : E.map f ≤ E) :
+  E.map f = E :=
+begin
+  haveI hE_submod_fin : finite_dimensional K E.to_subalgebra.to_submodule := h_findim,
+  have h_finrank_eq : finite_dimensional.finrank K E = finite_dimensional.finrank K (E.map f),
+  { exact (E.to_subalgebra.to_submodule.equiv_map_of_injective
+      f.to_linear_map f.to_ring_hom.injective).finrank_eq },
+  exact intermediate_field.to_subalgebra_eq_iff.mp (subalgebra.to_submodule_injective
+    (finite_dimensional.eq_of_le_of_finrank_eq h_map_le h_finrank_eq.symm)),
+end
+
+lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_surjective (h_int :
+  algebra.is_integral K L) (f : ultrafilter (L →ₐ[K] L)) :
+  function.surjective (ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int) :=
+begin
+  intro y,
+  let p := minpoly K y,
+  let S := p.root_set L,
+  let σ := ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int,
+  have hp : p ≠ 0 := (minpoly.monic (h_int y)).ne_zero,
+  have hσSS : σ '' S ⊆ S,
+  { rintros x ⟨a, ha, rfl⟩,
+    rw polynomial.mem_root_set hp at ha,
+    rw [polynomial.mem_root_set hp, polynomial.aeval_alg_hom_apply, ha, σ.map_zero] },
+  have h3 : σ '' S = S :=
+  eq_of_subset_of_card_le hσSS (ge_of_eq (card_image_of_injective S σ.to_ring_hom.injective)),
+  have hy : y ∈ σ '' S,
+  { rw [h3, polynomial.mem_root_set hp, minpoly.aeval] },
+  rcases hy with ⟨a, ha, hay⟩,
+  exact ⟨a, hay⟩,
+end
+
+lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_bijection (h_int : algebra.is_integral
+  K L) (f : ultrafilter (L →ₐ[K] L)) : function.bijective
```

Comment:
Line-wrapping in the middle of arguments tends not to be something we do if we can avoid it:
```suggestion
lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_bijection
  (h_int : algebra.is_integral K L) (f : ultrafilter (L →ₐ[K] L)) : function.bijective
```
Same throughout